### ISSUE TITLE
Simple associated product of configurable product removed from website generates cart error

### ISSUE THREAD

**Issue by daniel-ifrim** [magento/magento2#24928]
### Preconditions (*)
1. Magento 2.3.0. Didn't test on Magento 2.3.2

### Steps to reproduce (*)
1. Have 2 websites (as in website > store) configured in admin.
2. Have the simple associated product in both websites.
3. Login on frontend as user.
4. Add configurable product to cart, the swatch/option for associated simple product.
5. Go to cart page.
6. Go to admin and remove a website from the simple associated product.
7. Refresh cart page

### Expected result (*)
1. No PHP fatal error
2.

### Actual result (*)
1. Fatal error on cart page
```
Uncaught TypeError: Return value of Magento\ConfigurableProduct\Model\Product\Configuration\Item\ItemProductResolver::getChildProduct() must be an instance of Magento\Catalog\Model\Product, null returned in vendor/magento/module-configurable-product/Model/Product/Configuration/Item/ItemProductResolver.php:79
Stack trace:
vendor/magento/module-configurable-product/Model/Product/Configuration/Item/ItemProductResolver.php(51): Magento\ConfigurableProduct\Model\Product\Configuration\Item\ItemProductResolver-&gt;getChildProduct(Object(Magento\Quote\Model\Quote\Item\Interceptor))
vendor/magento/module-catalog/Model/Product/Configuration/Item/ItemResolverComposite.php(39): Magento\ConfigurableProduct\Model\Product\Configuration\Item\ItemProductResolver-&gt;getFinalProduct(Object(Magento\Quote\Model\Quote\Item\Interceptor))
vendo in <b>magento/module-configurable-product/Model/Product/Configuration/Item/ItemProductResolver.php</b> on line <b>79</b><br />
```

More than this, cart section is broken and can't be loaded. The error persists. Customer can't buy products anymore.

**Comment by hostep** [comment]
It looks like this was fixed in Magento 2.3.1 with this commit: https://github.com/magento/magento2/commit/e7d5aa8c8bb959b3d03c5d8f75ce3eee4727df00#diff-23fa97cc9e22ec55a57b2109b07f57c3L71 where the return type of that method was changed from `Product` to `?Product` thus allowing it to return `null` values.

Let me close this ticket for now. Thanks for the report though!

**Comment by daniel-ifrim** [comment]
@hostep Thank you
